test/test_unit_keystore_credentials_provider.py

```diff
@@ -103,3 +106,15 @@ def test_remove_credentials_raises_keyring_error_when_delete_password_raises_key
         credential_provider = KeystoreCredentialsProvider()
         with self.assertRaises(keyring.errors.KeyringError):
             credential_provider.remove_credentials(MockConjurrcData)
+
```

Comment:
extra line 🔥

Reply:
when would delete_password raise a keyring error? From my understanding it should never. In keystore_adapter we have
```
    def delete_password(cls, identifier, key):
        """
        Method for deleting a password in keyring
        """
        try:
            keyring.delete_password(identifier, key)
        except keyring.errors.PasswordDeleteError as password_error:
            raise KeyringAdapterDeletionError(f"Failed to delete key '{key}' for identifier "
                                              f"'{identifier}'") from password_error
        except Exception as exception:
            raise KeyringAdapterGeneralError(message=f"General keyring error has occurred "
                                                     f"(Failed to delete '{key}')'") from exception
``` 
So any keyring error should be caught in the Exception and the actual method should raise `KeyringAdapterGeneralError` no?

Reply:
we use `keyring.errors.KeyringError` to simulate Exception. we could use other exceptions as well

Reply:
sure but yes we should be catching those errors but we should be throwing `KeyringAdapterGeneralError` not `keyring.errors.KeyringError`